src/ucp/rndv/rndv_put.c

```diff
@@ -129,26 +129,33 @@ static size_t ucp_proto_rndv_put_common_pack_atp(void *dest, void *arg)
     return ucp_proto_rndv_pack_ack(pack_ctx->req, dest, pack_ctx->ack_size);
 }
 
+static size_t
+ucp_proto_rndv_put_common_ack_size(ucp_request_t *req, ucp_lane_map_t atp_map)
+{
+    const ucp_proto_rndv_put_priv_t *rpriv = req->send.proto_config->priv;
+
+    /* When we need to send multiple ATP messages: each will acknowledge 1 byte,
+       except the last ATP which will acknowledge the remaining payload size.
+       This is simpler than keeping track of how much was sent on each lane */
```

Comment:
AFAIU we need to ack each lane because you need to verify they all finished sending all the data.
anyway this logic was already in the code, I just refactored it out in order to use it in `ucp_proto_rndv_put_zcopy_reset`

Reply:
ok